test/client_test.dart

```diff
@@ -25,4 +29,68 @@ void main() {
     expect(session.accessToken,
         'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJhdXRoZW50aWNhdGVkIiwiZXhwIjoxNjExODk1MzExLCJzdWIiOiI0Njg3YjkzNi02ZDE5LTRkNmUtOGIyYi1kYmU0N2I1ZjYzOWMiLCJlbWFpbCI6InRlc3Q5QGdtYWlsLmNvbSIsImFwcF9tZXRhZGF0YSI6eyJwcm92aWRlciI6ImVtYWlsIn0sInVzZXJfbWV0YWRhdGEiOm51bGwsInJvbGUiOiJhdXRoZW50aWNhdGVkIn0.GyIokEvKGp0M8PYU8IiIpvzeTAXspoCtR5aj-jCnWys');
   });
+
+  test('signUp()', () async {
+    final response = await client.signUp(email, password);
+    final data = response.data;
+    final error = response.error;
+    expect(error, isNull);
+    expect(data.accessToken is String, true);
+    expect(data.refreshToken is String, true);
+    expect(data.user.id is String, true);
+  });
+
+  test('signIn()', () async {
+    final response = await client.signIn(email: email, password: password);
+    final data = response.data;
+    final error = response.error;
+    expect(error, isNull);
+    expect(data.accessToken is String, true);
+    expect(data.refreshToken is String, true);
+    expect(data.user.id is String, true);
+  });
+
+  test('Get user', () async {
+    final user = client.user();
+    expect(user.id is String, true);
+    expect(user.appMetadata['provider'], 'email');
+  });
+
+  test('Update user', () async {
+    final response =
+        await client.update(UserAttributes(data: {'hello': 'world'}));
+    final data = response.data;
+    final error = response.error;
+    expect(error, isNull);
+    expect(data.id is String, true);
+    expect(data.userMetadata['hello'], 'world');
+  });
+
+  test('Get user after updating', () async {
+    final user = client.user();
+    expect(user.id is String, true);
+    expect(user.userMetadata['hello'], 'world');
+  });
+
+  test('signOut', () async {
+    final res = await client.signOut();
+    expect(res.error, isNull);
+  });
+
+  test('Get user after logging out', () async {
+    final user = client.user();
+    expect(user, isNull);
+  });
+
+  test('signIn() with the wrong password', () async {
+    final res = await client.signIn(
+      email: email,
+      password: '${password}2',
+    );
+    final data = res.data;
+    final error = res.error;
+    expect(
+        error.message, 'No user found with that email, or password invalid.');
+    expect(data, isNull);
```

Comment:
Do you have any good idea not to hard code this error message?

Reply:
we can check that error.message is not null. Good enough